backends/s3/s3.go

```diff
@@ -51,6 +51,13 @@ type Options struct {
 	// CreateBucket tells us to try to create the bucket
 	CreateBucket bool `yaml:"create_bucket"`
 
+	// GlobalPrefix is a prefix applied to all operations, allowing work within a prefix
+	// seamlessly
+	GlobalPrefix string `yaml:"global_prefix"`
+
+	// Recursive can be enabled to make List operations recurse into nested prefixes
+	Recursive bool `yaml:"recursive"`
```

Comment:
I would like to make listing all filenames with any slashes the default, instead of requiring a per-backend opt-in.

So instead of `Recursive` we would provide an option to opt-out of the recursion,

Reply:
Below commit removes `Recursive` and replaces it with `PrefixFolders`.

`PrefixFolders` can be enabled to make List operations show nested prefixes as folders instead of recursively listing all contents of nested prefixes.

Tests updated accordingly.